src/common/common-thread.c

```diff
@@ -27,6 +27,18 @@ int COMMON_PREFIX (thread_join) (bson_thread_t thread)
 {
    return pthread_join (thread, NULL);
 }
+
+bool COMMON_PREFIX (mutex_is_locked) (bson_mutex_t *mutex)
+{
+#ifdef MONGOC_ENABLE_TESTING
```

Comment:
This is not necessary. If `MONGOC_ENABLE_TESTING` is not defined, then `MONGOC_TEST_ASSERT` expands to `((void) 0)`.

Arguably, `mutex_is_locked` may never be called outside of test assertions. But having it return  `true` when the testing macro is not defined seems like potentially surprising behavior.